_posts/2016-12-02-one-year-on-gcp.md

```diff
@@ -0,0 +1,62 @@
+
+In this blog post will focus on why we at Unacast chose GCP as our cloud provider, why it’s still a good fit for us, and a few lessons learned of using the platform for about a year.
+
+## Why Google Cloud Platform
+
+Today the natural choice of cloud is Amazon Web Services (AWS). And with good reason. AWS pioneered many of the great cloud services out there, S3, EC2, Lambda etc. It has as far as I know the longest list of cloud components you can use to build your platform <sup>1</sup>. And it’s battle tested at scale by Amazon, Netflix, AirBnB and many others. So why did we choose GCP instead?
+
+Actually, we didn’t. Unacast started out building its platform on a combination of Heroku and AWS. And after some fumbling, sessions of banging our heads against the wall, and some help from a consultant<sup>2</sup> we decided to try GCP. And with some effort and a lot of luck it turned out to be the right platform for us. The reason being two fold 1) GCPs big data capabilities and 2) it helps us minimise time used on operations.  
+
+There is no secret that Google knows how to handle large amounts of data. And many of the tools provided at GCP is designed for handling storing and processing big data. Tools like Dataflow, Pubsub, BigQuery, Datastore, and BigTable are really powerful tools for data management. GCP also as great environments for running services like App-, Cloud Engine and Dataflow helps us maximise the time used to build business critical features fast, rather than using developer time on keeping the lights on.
+
+## The Good Parts
+
+#### Pubsub
+
+Pubsub is a distributed publish/subscriber queue. It can be used to propagate messages between services, but at Unacast we've mostly used it to regulating back pressure. And it works great in scenarios where you want to buffer traffic/load between front-line API endpoints and sub stream services. We use this approach designing write-centric APIs that can handle large unpredictable spikes of requests. NB! Pubsub doesn’t provide any ordering guarantees, and it doesn’t provide any retention unless a subscription is created for a topic.
+
+#### BigQuery
+
+BigQuery is a great database for building analytics tools. Storing data is cheap and you only pay for querying and storing data. BigQuery is great because of its out of the box capabilities for querying large amounts of data really fast. To put things into perspective our unscientific tests shows that BigQuery can query 1GB of data as fast as 100GB (and probably even more). One thing to remember when using BigQuery is that it's an append-only database, meaning that you cannot delete single rows only tables<sup>3</sup>. Where Cassandra as row-level TTLs BigQuery ships with table-level TTLs. So implementing data retention has to be done differently and may not be straight forward if you're coming from a standard SQL perspective. 
```

Comment:
<bigq is shipped with> table-level TTLs, so implementing <...>. This might not be <straight fwd>